teams_proxy_endpoints.go

```diff
@@ -0,0 +1,117 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"net/http"
+	"time"
+)
+
+type TeamsProxyEndpointListResponse struct {
+	Response
+	ResultInfo `json:"result_info"`
+	Result     []TeamsProxyEndpoint `json:"result"`
+}
+type TeamsProxyEndpointDetailResponse struct {
+	Response
+	Result TeamsProxyEndpoint `json:"result"`
+}
+
+type TeamsProxyEndpoint struct {
+	ID        string     `json:"id"`
+	Name      string     `json:"name"`
+	IPs       []string   `json:"ips"`
+	Subdomain string     `json:"subdomain"`
+	CreatedAt *time.Time `json:"created_at,omitempty"`
+	UpdatedAt *time.Time `json:"updated_at,omitempty"`
+}
+
+// ProxyEndpoints returns all proxy endpoints within an account.
+//
+// API reference: https://api.cloudflare.com/#teams-proxy-endpoints-list-proxy-endpoints
+func (api *API) ProxyEndpoints(ctx context.Context, accountID string) ([]TeamsProxyEndpoint, ResultInfo, error) {
```

Comment:
we should prefix these methods with `Teams`/`Gateway` etc to somewhat namespace them until we revamp the SDKs.

Reply:
done